core/org.wso2.carbon.tomcat/src/main/java/org/wso2/carbon/tomcat/internal/CarbonTomcat.java

```diff
@@ -408,8 +411,18 @@ public void startConnectors(int portOffset) {
         Connector[] connectors = this.getService().findConnectors();
         for (Connector connector : connectors) {
             try {
-                int currentPort = connector.getPort();
-                connector.setPort(currentPort + portOffset);
+                String isRandomPort = System.getProperty("tomcat.random.port.enable");
```

Comment:
Ok, now I understod that this is because the default behavior should be to pick ports from the catalina-server.xml so this implementation is correct.